src/main/scala/devices/spi/SPIPhysical.scala

```diff
@@ -66,7 +66,7 @@ class SPIPhysical(c: SPIParamsBase) extends Module {
 
   //Making a delay counter for 'sample'
   val totalCoarseDel = io.ctrl.extradel.coarse + io.ctrl.sampledel.sd
-  val del_cntr = RegInit(UInt(c.divisorBits.W), (4.U))
+  val del_cntr = RegInit(UInt(c.divisorBits.W), (c.defaultSampleDel + 1.U))
```

Comment:
I am not sure the order of operations here, for clarify can you make it `(c.defaultSampleDel + 1).U`? The math should be done by scala, not hardware

Reply:
okay, makes sense. Changed the logic.